src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests2.cs

```diff
@@ -954,6 +954,79 @@ public static void Main()
 9");
         }
 
+        [Fact]
+        public void TypeCheckInPropertyPattern()
+        {
+            var source =
+@"using System;
+
+class Program2
+{
+    public static void Main()
+    {
+        object o = new Frog(1, 2);
+        if (o is Frog(1, 2))
+        {
+            Console.Write(1);
+        }
+        if (o is Frog { A: 1, B: 2 })
+        {
+            Console.Write(2);
+        }
+        if (o is Frog(1, 2) { A: 1, B: 2, C: 3 })
+        {
+            Console.Write(3);
+        }
+    }
+}
+
+class Frog
+{
+    public object A, B;
+    public object C => (int)A + (int)B;
+    public Frog(object A, object B) => (this.A, this.B) = (A, B);
+    public void Deconstruct(out object A, out object B) => (A, B) = (this.A, this.B);
+}
+";
+            var compilation = CreatePatternCompilation(source);
+            compilation.VerifyDiagnostics(
+                );
+            var comp = CompileAndVerify(compilation, expectedOutput: @"123");
+        }
+
+        [Fact]
+        public void OvereagerSubsumption()
```

Comment:
>OvereagerSubsumption [](start = 20, length = 20)

Does this test involve subsumption?

Reply:
No, it involves error recovery, which previously improperly interacted with subsumption (an erroneous pattern was treated as subsuming a non-erroneous one).

---
In reply to: [176464065](https://github.com/dotnet/roslyn/pull/25544#discussion_r176464065) [](ancestors = 176464065)